openfunctions/README.md

```diff
@@ -145,6 +145,170 @@ output = pipe(prompt)
 print(output)
 ```
 
+## Self-Hosting OpenFunctions
+
+This section provides a guide on how to self-host the OpenFunctions model on your local machine or serve it locally for your enterprise. The server deploys the OpenFunctions-v0 model with uvicorn, while the client interacts with this local server using the OpenAI package (0.28.xx). 
+
+### Setting Up Your Local Server
+
+The server API endpoint mirrors the interface of the API call executed by `openai.ChatCompletion.create`, ensuring compatibility with clients using the OpenAI package.
+
+Ensure you have the required libraries:
+```bash
+!pip install fastapi uvicorn transformers torch
+```
+
+Similar to the process outlined in the "Running OpenFunctions Locally" section, the model is loaded from HuggingFace for local serving. Only use the `get_prompt` format if you are hosting locally.
```

Comment:
tested on flaminio, works! Could you add a link to the colab notebook in the README, and add the ngrok instructions on the colab notebook? I think it would be useful for others. Specify that you need the A100 instance though; T-4 is too small as it ooms.

Reply:
Thanks for the feedback! I added a link to the Colab notebook to the README, specified minimum GPU requirements, and added instructions for setting up ngrok to the notebook.